pkg/validator/cluster.go

```diff
@@ -776,6 +781,57 @@ func validateSecrets(
 	return valErrors, patches
 }
 
+// encryptSecretKeys encrypts secret keys per each role and generates patches if needed
+func encryptSecretKeys(cr *kdv1.KubeDirectorCluster, prevCr *kdv1.KubeDirectorCluster, valErrors []string, patches []clusterPatchSpec) ([]string, []clusterPatchSpec) {
+	for roleIndex, role := range cr.Spec.Roles {
+		var needsUpdate bool
+		for keyIndex, secretKey := range role.SecretKeys {
+			prevSecretKey := findSecretKey(prevCr, role.Name, secretKey.Name)
+			encryptedSecretKey := secretKey
+			if prevSecretKey != nil && secretkeys.Equal(*prevSecretKey, secretKey) {
+				encryptedSecretKey = *prevSecretKey
+			}
+			encryptedSecretKey, err := secretkeys.EncryptSecretKey(encryptedSecretKey)
+			if err != nil {
+				// TODO Better error handling
```

Comment:
I've been thinking if it should log encryption error to KD logs

Reply:
Might as well yep. As long as we don't log any actual keys.

Reply:
Yeah, I've been struggling on how to obtain logger in that function to do that